src/javascript/javascript-import-scanner.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// import * as estraverse from 'estraverse';
+import * as estree from 'estree';
+import {resolve as resolveUrl} from 'url';
+
+import {ScannedImport} from '../ast/ast';
+import {Visitor} from '../javascript/estree-visitor';
+import {JavaScriptDocument} from '../javascript/javascript-document';
+import {JavaScriptScanner} from '../javascript/javascript-scanner';
+
+export class JavaScriptImportScanner implements JavaScriptScanner {
+  async scan(
+      document: JavaScriptDocument,
+      visit: (visitor: Visitor) => Promise<void>): Promise<ScannedImport[]> {
+    const imports: ScannedImport[] = [];
+
+    await visit({
+      enterImportDeclaration(node: estree.ImportDeclaration, _: estree.Node) {
+        const source = node.source.value as string;
+        if (!isPathImport(source)) {
+          // TODO(justinfagnani): push a warning
```

Comment:
Oh, oops, did mean to skip it. I'll add a test.

We do need a way for a scanner to report back errors/warnings though.